### DISCUSSION TITLE
Return a CSV instead of JSON

### DISCUSSION THREAD

**Discussion by twiclo** [graphql-rust/juniper#1163] [Q&A] [answered]
I need to return the results of a graphql query as a csv instead of a json. I was thinking about having my frontend send the query to a REST endpoint, the function would run it through juniper, and respond with a CSV. Can the devs give input on a slicker way of pulling this off?

**Comment by LegNeato** [comment]
There is nothing json-specific about juniper, so you can do whatever you want. The built integrations with http server frameworks use JSON, but there's no reason why you can't wrap them and use CSV for whatever logic you want. 

Or what its worth, The most idiomatic way is to have some sort of CSV field that you query via GraphQL and is encoded by a JSON. But again JSON isn't a requirement for GraphQL or juniper

**Reply by twiclo** [reply]
Well how do you do that? If I run a query through `execute()` it returns a GraphQLResponse. I've been wrestling with getting one of these into a csv format with little luck

**Reply by tyranron** [reply, marked as answer]
@twiclo thus, there is no way to expose the inner `juniper::Value` from `GraphQLResponse`, the latest implements `serde::Serialize`. So, you either need a `serde`-compatible CSV serializer, or, at least, serialize it into a `serde_json::Value` (or similar) and then use it for your custom serialization, which won't differ much from serializaing `juniper::Value` directly.
